nixos/modules/services/databases/postgresql.nix

```diff
@@ -565,7 +583,11 @@ in
               )
               cfg.ensureUsers
             }
-          '';
+          '' + concatMapStrings (user: let
+            userRoles = concatStringsSep "," user.ensureRoles;
+          in ''
+            $PSQL -tAc 'GRANT "${userRoles}" TO "${user.name}"'
+          '') (filter (user: length user.ensureRoles != 0) cfg.ensureUsers);
```

Comment:
AFAIU GRANTs aren't dropped here when removed from the configuration, right?
I'm not sure if there's a reasonable way to implement that, but if not then it should be warned about this behavior in at least the option's description.

Reply:
> AFAIU GRANTs aren't dropped here when removed from the configuration, right?

Correct.

> I'm not sure if there's a reasonable way to implement that, but if not then it should be warned about this behavior in at least the option's description.

There's not, we're dealing with an inherently stateful system. Nix can only do so much in that regard.

I had planned to add it to the description, I just wanted to nail down what we actually want with this PR before I continued.